src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,784 @@
+        Expected (After truncation): 555 installation patches in status
+        operation: Installation,
+        installation substatus name: PatchInstallationSummary,
+        installation substatus status: warning,
+        installation errors code: 0 (success),
+        installation errors details count: 0,
+        count of installation patches removed: 99445,
+        truncated status file byte size: 126kb. """
+
+        self.runtime.execution_config.operation = Constants.INSTALLATION
+        self.runtime.status_handler.set_current_operation(Constants.INSTALLATION)
+        # Set up tmp file store all log/print message and set sys.stdout point to it.
+        # Note: sys log/print will not display in terminal, for debugging comment __remove_tmp_file_reset_stdout and check the tmp file
+        self.__create_tmp_file_for_log_and_set_stdout()
+
+        patch_count = 100000
+        self.__expected_truncated_patch_count = 555
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions, Constants.INSTALLED)
+        self.runtime.status_handler.set_installation_substatus_json(status=Constants.STATUS_SUCCESS)
+        self.runtime.status_handler.log_truncated_patches()
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)[0]
+
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY,
+            Constants.STATUS_SUCCESS, patch_count)
+
+        # Assert installation truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            truncated_substatus_file_data = json.load(file_handle)[0]
+
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY,
+            Constants.STATUS_WARNING, patch_count, complete_substatus_file_data=complete_substatus_file_data, is_under_internal_size_limit=True, is_truncated=True)
+
+        # Assert all installation fields in the message json are equal in both status files
+        self.__assert_installation_truncated_msg_fields(complete_substatus_file_data, truncated_substatus_file_data)
+
+        # Assert 'Count of patches removed from: [Assessment=0] [Installation=xxx] log message is called
+        self.__assert_truncated_patches_removed(truncated_substatus_file_data, patch_count_installation=patch_count)
+
+        # Reset sys.stdout, close and delete tmp file
+        # Note: sys log/print will not display in terminal, for debugging comment __remove_tmp_file_reset_stdout and check the tmp file
+        self.__remove_tmp_file_reset_stdout()
```

Comment:
To be removed

Applies in all other tests as well

Reply:
logic is removed

done.